File: BasicOperators/src/main/java/suppressing_operators/filter/Launcher.java
package suppressing_operators.filter;

import io.reactivex.Observable;


public class Launcher {
    public static void main(String[] args) {
        Observable.just("Alpha", "Beta", "Gamma", "Delta", "Epsilon")
                .filter(s -> s.length() != 5)
                .subscribe(s -> System.out.println("RECEIVED: " + s));
    }
}

File: ObserversSubscribers/hola/src/main/java/Launcher.kt
import io.reactivex.Observable
import java.util.concurrent.TimeUnit

fun main() {
    val myStrings: Observable<String> = Observable.just("Alpha", "Beta", "Gamma", "Delta", "Epsilon")
    myStrings
            .doOnTerminate {
                println("do Onterminate")
            }

            .doOnNext {
                println("doOnnext: $it")
            }
            .doOnComplete {
                println("doOncomplet")
            }
            .map {
                it.length
            }
            .map {
                it.plus(it).equals(10)
            }
            .subscribe {
                println(it)
            }
}

